### ISSUE TITLE
`hxlquickimporttab`

### ISSUE THREAD

**Issue by fititnt** [EticaAI/HXL-Data-Science-file-formats#7]
TODO: add context

**Comment by fititnt** [comment]
Humm... that's interesting. The hxlquickimport (since the underlining libhxl already auto discover that is a tab-like CSV) almost work. It does not break the the classes further, and would require human correction, but I think the importer could eventually include extra check if is an .tab.

Anyway, the idea of hxlquickimporttab is somewhat be able to both import already exported .tab files based on HXL, and (this would need more test cases) ones that are more generic (but still ones generated by Orange Canvas).


```bash
fititnt@bravo:/workspace/git/EticaAI/HXL-Data-Science-file-formats$ head tests/files/iris_hxlated-tab.tab 
C#item+eng_sepal+eng_length+number	C#item+eng_sepal+eng_width+number	C#item+eng_petal+eng_length+number	C#item+eng_petal+eng_width+number	c#item+class+vt_class
5.1	3.5	1.4	0.2	Iris-setosa
4.9	3	1.4	0.2	Iris-setosa
4.7	3.2	1.3	0.2	Iris-setosa
4.6	3.1	1.5	0.2	Iris-setosa
5	3.6	1.4	0.2	Iris-setosa
5.4	3.9	1.7	0.4	Iris-setosa
4.6	3.4	1.4	0.3	Iris-setosa
5	3.4	1.5	0.2	Iris-setosa
4.4	2.9	1.4	0.2	Iris-setosa
fititnt@bravo:/workspace/git/EticaAI/HXL-Data-Science-file-formats$ hxlquickimporttab tests/files/iris_hxlated-tab.tab | head
#item+c_item_eng_sepal_eng_length_number,#item+c_item_eng_sepal_eng_width_number,#item+c_item_eng_petal_eng_length_number,#item+c_item_eng_petal_eng_width_number,#item+c_item_class_vt_class
5.1,3.5,1.4,0.2,Iris-setosa
4.9,3,1.4,0.2,Iris-setosa
4.7,3.2,1.3,0.2,Iris-setosa
4.6,3.1,1.5,0.2,Iris-setosa
5,3.6,1.4,0.2,Iris-setosa
5.4,3.9,1.7,0.4,Iris-setosa
4.6,3.4,1.4,0.3,Iris-setosa
5,3.4,1.5,0.2,Iris-setosa
4.4,2.9,1.4,0.2,Iris-setosa
```

**Comment by fititnt** [comment]
The next hxl2tab (#2), will be less verbose. But in short, while it will keep on the hashtag+attibutes the more generic +text, +number, #meta ... the very specific `+vt_orange_flag_class`, `+vt_class`, `+vt_class`, `+vt_categorical` would be redundant.

The `hxlquickimporttab`, different from the hxlquickimport, really needs to do some check if the .tab already is an HXLated tab. **And that's actually very important**, because the import would be different. This check needs to be implemented, even if basic.

The current versions output

```bash
(base) fititnt@bravo:/workspace/git/EticaAI/HXL-Data-Science-file-formats$ hxl2tab https://docs.google.com/spreadsheets/d/1Vqv6-EAdSHMSZvZtE426aXkDiwP8Mdrpft3tiGQ1RH0/edit#gid=0 | head
cD#status	D#country	D#adm1	D#adm1+code	mD#loc	mD#loc	mD#org	m#loc+type	C#affected+dead+number	C#affected+confirmed+number	C#affected+suspected+number
Pending	Liberia	Margibi	LR09	Kakata 1	Kakata 2 AFL	AFL	ETC	0	0	0
Functional	Guinea	Nzerekore	GN008		Nzerekore	Ailema (?)	ETC	45	56	3
Pending	Liberia	River Gee	LR13	Fishtown	Fishtown ETC	American Red Cross	ETC	0	0	0
Functional	Sierra Leone	Western	SL04	Jui	Sierra Leone-China Friendship Hospital (Jui Hospital)	Chinese CDC	ETC	47	65	17
Pending	Guinea	Nzerekore	GN008			Croix-Rouge franÃ§aise	ETC	0	0	0
Pending	Sierra Leone	Western	SL04	Freetown	Goderich	EMERGENCY	ETC	0	0	0
Functional	Sierra Leone	Western	SL04	Lakka	Lakka Hospital ETU	EMERGENCY Italian NGO	ETC	3	17	11
Functional	Liberia	Margibi	LR09	Firestone	Firestone Medical Center	Firestone Company	ETC	14	29	19
Functional	Liberia	Montserrado	LR11	Monrovia	Monrovia, Congo Town - Old Ministry of Defence ETU 1	FMT	ETC	1	30	6
(base) fititnt@bravo:/workspace/git/EticaAI/HXL-Data-Science-file-formats$ hxl2tab https://docs.google.com/spreadsheets/d/1Vqv6-EAdSHMSZvZtE426aXkDiwP8Mdrpft3tiGQ1RH0/edit#gid=0 | hxlquickimporttab | head
['#item+cd_status', '#item+d_country', '#item+d_adm1', '#item+d_adm1_code', '#item+md_loc', '#item+md_loc', '#item+md_org', '#item+m_loc_type', '#item+c_affected_dead_number', '#item+c_affected_confirmed_number', '#item+c_affected_suspected_number']
#item+cd_status,#item+d_country,#item+d_adm1,#item+d_adm1_code,#item+md_loc,#item+md_loc,#item+md_org,#item+m_loc_type,#item+c_affected_dead_number,#item+c_affected_confirmed_number,#item+c_affected_suspected_number
Pending,Liberia,Margibi,LR09,Kakata 1,Kakata 2 AFL,AFL,ETC,0,0,0
Functional,Guinea,Nzerekore,GN008,,Nzerekore,Ailema (?),ETC,45,56,3
Pending,Liberia,River Gee,LR13,Fishtown,Fishtown ETC,American Red Cross,ETC,0,0,0
Functional,Sierra Leone,Western,SL04,Jui,Sierra Leone-China Friendship Hospital (Jui Hospital),Chinese CDC,ETC,47,65,17
Pending,Guinea,Nzerekore,GN008,,,Croix-Rouge franÃ§aise,ETC,0,0,0
Pending,Sierra Leone,Western,SL04,Freetown,Goderich,EMERGENCY,ETC,0,0,0
Functional,Sierra Leone,Western,SL04,Lakka,Lakka Hospital ETU,EMERGENCY Italian NGO,ETC,3,17,11
Functional,Liberia,Margibi,LR09,Firestone,Firestone Medical Center,Firestone Company,ETC,14,29,19
```

**Comment by fititnt** [comment]
The binary exist at `bin/hxlquickimporttab`, but while inspired other tooling, (first **HXLTabConverter common class #8**, then **[meta] HDP Declarative Programming (working draft) #16**, much, much more complex), it was not documented.

I will close this issue, but the **hxlquickimporttab** is unlikely to receive updates, since part of functionality was replaced either by #8 or the **hxlquickimport #6**, which is documented.

In other words: the hxlquickimporttab is a legacy program. Most features now allow customization using YAML, and it was from the time it required know python.
